### ISSUE TITLE
Each grade value has no parent concept. Should we connect or create a hierarchy above it?

### ISSUE THREAD

**Issue by rtmill** [OHDSI/OncologyWG#558]

**Comment by golozara** [comment]
@kzollove

**Comment by kzollove** [comment]
[original top comment from @rtmill]
And if so, should that also go a level or two higher and create a more comprehensive set of staging/grading concepts? 

See: #519

**Comment by rtmill** [comment]
Pasting comment from duplicate ticket before closing it: 

"
[rtmill](https://github.com/rtmill) commented [on Oct 30, 2023](https://github.com/OHDSI/OncologyWG/issues/569#issuecomment-1785577236)
The issue: The values within Staging and grading are the top of their respective hierarchies. To make it searchable and easy to navigate, should we insert relations to parent concepts to group them?

Examples:
[Grade 1 tumor](https://athena.ohdsi.org/search-terms/terms/1634371) has no parent concept
[TNM and Stage Group](https://athena.ohdsi.org/search-terms/terms/734320) both have the highest level concept as "AJCC/UICC finding"

The goal would be to have concepts a user could click on and then navigate all standard concepts that fit within "grade", "stage", etc, as right now there is no grouper for them.

Option A) Create new concepts within Cancer Modifier to encapsulate these groups
Option B) Leverage a grouper concept from another vocabulary (e.g. SNOMED), assuming all child-concepts are de-duplicated
Option C) Another plan
"

**Comment by kzollove** [comment]
These two CSV files contain the current "top dog" concept and classification values in the staging/grading concept_class of OMOP vocab.

Part of this proposal is that none of these concept and classification values should be at the top of a hierarchy; they all should be further classified.

That said, the list of concept values could likely be scrutinized even further. Its possible some of these concepts could be removed or better integrated into the hierarchy.

[top_level_concepts_staging_grading.csv](https://github.com/OHDSI/OncologyWG/files/14564389/top_level_concepts_staging_grading.csv)
[top_level_classifications_staging_grading.csv](https://github.com/OHDSI/OncologyWG/files/14564390/top_level_classifications_staging_grading.csv)

**Comment by kzollove** [comment]
#### August 17 2023 Proposal (replaced in top comment with a more detailed proposal and moved here)

**TLDR; We should create terms Tumor Finding, Tumor Stage Finding, and Tumor Grade Finding to create unifying "top dog" concepts for the Stage and Grade hierarchies in Cancer Modifier vocabulary.**

![Image](https://github.com/OHDSI/OncologyWG/assets/18039765/9008d62a-a20d-46af-9f05-be9303d0ff18)

## Background 
In SNOMED, the concept Tumor Finding exists as a child of Finding of Lesion. Tumor Finding is a junkyard (catch-all) for concepts ranging from [Signet-ring cells present, comprising less than 50 percent of malignant cells](https://athena.ohdsi.org/search-terms/terms/4311712), Renal Tumor finding, [Tumor infiltration by lymphocytes brisk](https://athena.ohdsi.org/search-terms/terms/4245944), [Tumor involves both ovaries diffusely, primary tumor site cannot be determined](https://athena.ohdsi.org/search-terms/terms/4227239), and more. One of Tumor Finding’s children is the term Tumor Stage Finding. Tumor Stage Finding’s children are a bit less eclectic but also cover quite a broad scope such as terms that cover histology, topography, and stage (e.g. [Adenocarcinoma of lung, stage II](https://athena.ohdsi.org/search-terms/terms/4314832), [Carcinoma of ovary, stage 1](https://athena.ohdsi.org/search-terms/terms/4311576)), tumor stages ([Clinical stage I](https://athena.ohdsi.org/search-terms/terms/443907), [T0 category](https://athena.ohdsi.org/search-terms/terms/4243213)) and other classifiers ([TNM tumor staging finding](https://athena.ohdsi.org/search-terms/terms/4295649))

## Proposal: 

Introduce a “Tumor Finding” hierarchy to Cancer Modifier. This top-level concept should be parent to “Tumor Stage Finding”, “Tumor Grade Finding”, and other Tumor observation/modifiers. These concepts are useful for grouping together various tumor findings/observations/modifiers. This is a fairly non-controversial move that would have no effect on the current meaning or use any children concepts, but is crucial step in organizing these Cancer Modifier terms in a logical hierarchical way.

**Comment by rtmill** [comment]
In an attempt to be comprehensive... ( should we decide to separate this out into new discussions happy to do so, but these seemingly fit within the scope of this overhaul )

Adding to the proposal, as they have direct relevance here (and to put a nice bow on this staging/grading hierarchy rehaul altogether)

(big thanks @gkennos for investigating, working through this with me, and providing lists of concepts to be mapped) 


These two additions are very similar to one another and would exist within the same level of the same hierarchy, to which I will elaborate on at the end. 

----

**1)** **Add two parent CLASSIFIER concepts for Clin and Path within AJCC** (and create parent relationships to these classifiers for applicable child concepts) 

See:  #641 

Problem: 
- The specification of C vs. P is embedded within various hierarchies and yet not it's on hierarchy, i.e., differentiating between the two is not currently searchable/queryable in a comprehensive way (without needing to list all of the concepts or do a regex) 

Use case: 
- "limit to only cases where diagnosis was stated as pathologically confirmed", etc. 

Solution: 
- Create two new CLASSIFIER concepts, meant to simply the search and specification of C vs. P
  - AJCC/UICC **Clinical** Finding
  - AJCC/UICC **Pathological** Finding
- Create mapping from top level of each existing C or P hierarchy to the appropriate PARENT CLASSIFIER 

Given in AJCC/UICC that there are already multiple, overlapping hierarchies at a specific level in the hierarchy, I'd argue it makes sense to insert these two parents as children of [AJCC/UICC finding](https://athena.ohdsi.org/search-terms/terms/734320) - similar to how AJCC version and stage value group also serve as classifiers at this level (see subsumes section of previous link).

----

**2)  Add three parent CLASSIFIER concepts for T, N, and M measurements** (and create parent relationships to these classifiers for applicable child concepts). This is nearly identical in terms of the problem being solved and how it is being solved as #1 above) 

Problem:
- In order to search or query for all permutations of T, N or M, you currently need to explicitly list out all of the codes and cannot do through standard vocabulary querying via ancestor concepts

Use case: 
- "find all T values for this patient" , etc.

Solution: 
- Create three new **CLASSIFIER** concepts: 
  - AJCC/UICC **T** Finding 
  - AJCC/UICC **N** Finding 
  - AJCC/UICC **M** Finding 
- Create mapping from top level of each existing C or P hierarchy to the appropriate PARENT CLASSIFIER 


Given in AJCC/UICC that there are already multiple, overlapping hierarchies at a specific level in the hierarchy, I'd argue it makes sense to insert these two parents as children of [AJCC/UICC finding](https://athena.ohdsi.org/search-terms/terms/734320) - similar to how AJCC version and stage value group also serve as classifiers at this level (see subsumes section of previous link).

Specifically, these new parent concepts would be inserted at that same level, and then the existing T, N and M grouping concepts would be moved down a level, into their respective categories with these new classifier concepts as parents. 

----
**Two addendums summarized together**

The current highest level structure of AJCC/UICC Finding currently looks like this: 
(subsumes section of [AJCC/UICC Finding](https://athena.ohdsi.org/search-terms/terms/734320) 

CURRENT: 
| concept_name | concept_id | vocabulary_id 
-- | -- | --
   AJCC/UICC 6th edition | 1634647 | Cancer Modifier
   AJCC/UICC 7th edition | 1633496 | Cancer Modifier
   AJCC/UICC 8th edition | 1634449 | Cancer Modifier
   **AJCC/UICC M0 Category** | 1635624 | Cancer Modifier
   **AJCC/UICC M1 Category** | 1635142 | Cancer Modifier
   **AJCC/UICC MX Category** | 1633547 | Cancer Modifier
   **AJCC/UICC N0 Category** | 1633440 | Cancer Modifier
   **AJCC/UICC N1 Category** | 1634434 | Cancer Modifier
   **AJCC/UICC N2 Category** | 1634119 | Cancer Modifier
   **AJCC/UICC N3 Category** | 1635320 | Cancer Modifier
   **AJCC/UICC N4 Category** | 1635445 | Cancer Modifier
   **AJCC/UICC NX Category** | 1633885 | Cancer Modifier
   AJCC/UICC Stage 0 | 1633754 | Cancer Modifier
   AJCC/UICC Stage 1 | 1633306 | Cancer Modifier
   AJCC/UICC Stage 2 | 1634209 | Cancer Modifier
   AJCC/UICC Stage 3 | 1633650 | Cancer Modifier
   AJCC/UICC Stage 4 | 1633308 | Cancer Modifier
   **AJCC/UICC T0 Category** | 1634213 | Cancer Modifier
   **AJCC/UICC T1 Category** | 1635564 | Cancer Modifier
   **AJCC/UICC T2 Category** | 1635562 | Cancer Modifier
   **AJCC/UICC T3 Category** | 1634376 | Cancer Modifier
   **AJCC/UICC T4 Category** | 1634654 | Cancer Modifier
   **AJCC/UICC TX Category** | 1635682 | Cancer Modifier
   **AJCC/UICC Ta Category** | 1635114 | Cancer Modifier
   **AJCC/UICC Tis Category** | 1634530 | Cancer Modifier

Instead, if we make the above changes... 

PROPOSED:

| concept_name | concept_id | vocabulary_id 
-- | -- | --
   AJCC/UICC 6th edition | 1634647 | Cancer Modifier
   AJCC/UICC 7th edition | 1633496 | Cancer Modifier
   AJCC/UICC 8th edition | 1634449 | Cancer Modifier
   **AJCC/UICC T Finding** | 77777777777 | Cancer Modifier 
   **AJCC/UICC M Finding** | 888888888888 | Cancer Modifier
   **AJCC/UICC N Finding** | 999999999999 | Cancer Modifier
   AJCC/UICC Stage 0 | 1633754 | Cancer Modifier
   AJCC/UICC Stage 1 | 1633306 | Cancer Modifier
   AJCC/UICC Stage 2 | 1634209 | Cancer Modifier
   AJCC/UICC Stage 3 | 1633650 | Cancer Modifier
   AJCC/UICC Stage 4 | 1633308 | Cancer Modifier
   **AJCC/UICC Clinical Finding** | 888888888888 | Cancer Modifier
  **AJCC/UICC Pathological Finding** | 999999999999| Cancer Modifier


It is also worth noting that the lower level concepts are already connected to parent, grouper concepts... 
Example that covers both addendums : T and Pathological
![image](https://github.com/OHDSI/OncologyWG/assets/13469976/19f48ae4-7d2a-4a0e-9ddd-c06ee5d2ed6f)

Above, you can see parent relationships to grouper concepts, but the issue is that those grouper concepts themselves are not encompassed. (i.e. the concept 'AJCC/UICC pathological T2d Category' lacks both an overall 'Pathological Finding' ancestor as well as a 'T Finding' ancestor). This gap consequently requires enumerating those lists of grouper concepts instead of being able to leverage a single ancestor concept to conduct your queries. 

In other words, **we are not altering any existing functionality**, only adding new mechanisms to more efficiently search and query the discrete sets of standard concepts. The issue we are fixing is that those parent concepts themselves lack higher level, ancestor concepts.

--- 

To beat a dead horse, an example from each proposal, both child concepts of 'AJCC/UICC Finding', with their respective child (subsumes) concepts:

a) classifier concept : **AJCC/UICC T Finding**

Subsumes:
| concept_name | concept_id | vocabulary_id 
-- | -- | --
   **AJCC/UICC T0 Category** | 1634213 | Cancer Modifier
   **AJCC/UICC T1 Category** | 1635564 | Cancer Modifier
   **AJCC/UICC T2 Category** | 1635562 | Cancer Modifier
   **AJCC/UICC T3 Category** | 1634376 | Cancer Modifier
   **AJCC/UICC T4 Category** | 1634654 | Cancer Modifier
   **AJCC/UICC TX Category** | 1635682 | Cancer Modifier
   **AJCC/UICC Ta Category** | 1635114 | Cancer Modifier
   **AJCC/UICC Tis Category** | 1634530 | Cancer Modifier

a) classifier concept : **AJCC/UICC Pathological Finding**

(note the 'p-' at the beginning of each concept_code, a convention for classifying as pathological)

Subsumes:
| concept_name | concept_id 
-- | -- | --
p-AJCC/UICC-M0 | 1634618
p-AJCC/UICC-M1 | 1635505
p-AJCC/UICC-MX | 1633421
p-AJCC/UICC-N0 | 1635597
p-AJCC/UICC-N1 | 1635613
p-AJCC/UICC-N2 | 1633864
p-AJCC/UICC-N3 | 1635706
p-AJCC/UICC-N4 | 1634916
p-AJCC/UICC-NX | 1635170
p-AJCC/UICC-Stage-0 | 1633542
p-AJCC/UICC-Stage-1 | 1634252
p-AJCC/UICC-Stage-2 | 1633702
p-AJCC/UICC-Stage-3 | 1635566
p-AJCC/UICC-T0 | 1635740
p-AJCC/UICC-T1 | 1634004
p-AJCC/UICC-T2 | 1633978
p-AJCC/UICC-T3 | 1634406
p-AJCC/UICC-T4 | 1633943
p-AJCC/UICC-TX | 1633925
yp-AJCC/UICC-M0 | 1633364
yp-AJCC/UICC-M1 | 1634355
yp-AJCC/UICC-MX | 1635235
yp-AJCC/UICC-N0 | 1633527
yp-AJCC/UICC-N1 | 1634788
yp-AJCC/UICC-N2 | 1635061
yp-AJCC/UICC-N3 | 1635094
yp-AJCC/UICC-N4 | 1634466
yp-AJCC/UICC-NX | 1635503
yp-AJCC/UICC-Stage-0 | 1635675
yp-AJCC/UICC-Stage-1 | 1635095
yp-AJCC/UICC-Stage-2 | 1635689
yp-AJCC/UICC-Stage-3 | 1633837
yp-AJCC/UICC-T0 | 1634594
yp-AJCC/UICC-T1 | 1635781
yp-AJCC/UICC-T2 | 1634660
yp-AJCC/UICC-T3 | 1633436
yp-AJCC/UICC-T4 | 1635082
yp-AJCC/UICC-TX | 1633338

--- 

Appendix:

The benefits of this upgrade to this vocabulary seem clear. Any downsides, with the exception of needing to the do the work,  are not currently seen.

**Comment by vladkorsik** [comment]
Hi @rtmill, I'd like to carry on the conversation you initiated here.

**Disclaimer:**
Staging/Grading concept_class_id is a composite one entity that represents a collection of smaller findings regarding tumor behavior/biology - each concept is a small phenotype in itself. This concept class includes not only staging and grades but also other tumor attributes such as treatment responses. The proposed solution/hierarchy will cover only stages/grades per se. 

Here is what we see as possible pros and cons of the proposal:
**Pros**:
A way to systematically control "children"
**Cons**:

- Such high-level classifiers still won't be used in phenotypes. Creating them may make the impression that there exists a common entity to be used in phenotyping by collecting all its children in a concept set, but that is not the case.
- It's not very clear how this will actually help in the Athena search for ETL purposes; it still requires knowledge of the naming of “S” concepts. Moreover, concept_codes of the staging/grading concepts in Cancer Modifier already depict the **classification** that the code pertains to (e.g. c-6th_**AJCC/UICC**-M0 for AJCC/UICC 6th clinical M0 Category). Indirect querying through concept_ancestor/concept_relationship  will affect performance compared to direct querying through concept_code.
- The management of vocabulary will become more intricate, particularly considering the semi-manual approach to concept construction.

**Discussion:**
Preserving the staging or grading type of semantics as a concept attribute, particularly under concept_class, would be advantageous. The concept_class for Staging/Grading may be retained for 'C' concepts with overlapping coverage, as per [FIGO finding](https://athena.ohdsi.org/search-terms/terms/734316), thus it can remove the need to create “Finding-category - specific alternatives”. This way, the vocabulary will stay clean of concepts created just for patch-like reasons. 
Also, concept_class_id is a filtering field in OHDSI Tools like Atlas, so the metadata level manipulations may be beneficial.

**Comment by rtmill** [comment]
Thank you @vladkorsik for the thought out response. 

The "ATHENA use case" justification was admittedly confusing but hopefully I can clarify what we meant. 

I agree with you that concept_class is something we should (rework and) utilize better in this area.  The reason we did not address that here was to try and limit the (already large) scope of this specific decision point. As you stated, having appropriate concept_class assignments will make the navigation of concepts in ATHENA and similar tools easier, but we are hoping to do more than that. Additionally, I would argue that treatment response should not exist within Staging/Grading (see below), given that's something else entirely.  

Suggestions for concept_class modifications (if we want to include in this decision point): 
- Getting rid of the Staging/Grading class and instead creating...
  - Staging concept_class
  - Grading concept_class
  - Tumor progression finding concept_class
  - (and creating additional classes for the other tumor finding subcategories as we discuss and determine the discrete set, such as tumor dimension, disease status finding, etc.)

That said, the instantiation of these new **concept classes would likely be mostly only for navigational utility** - but - after the amount of work it took to track down the relevant concepts for each group in the current structure while putting this proposal together, they would be great to have and likely cause less confusion by not bundling them all together. 

I also agree that this proposal would make this portion of the Cancer Modifier vocabulary more "intricate", as you said, but I also believe that it will make it more **extensible and intuitive,** especially given that this is a custom vocabulary. The general overall goal is to have a 'Tumor Finding' hierarchy (as seen in diagrams above) and be able to easily query (i.e. without needing to list out every specific concept or use regex) for various subtypes of findings. 

I also agree that a portion of these sub-hierarchies are not likely to be used in phenotyping, and instead for other purposes such as **data characterization** and determining if an OMOP instance is **fit for use** for a given study. Given the variation in levels of detail from each site (e.g. excluding sites that have diagnoses but no findings), these sorts of things seem worthy of being able to determine in an easy and transparent way. 

e.g. if a study required a certain number of patients with specific types of tumor findings:
- Number of patients with cancer diagnosis that also have:
  - Any tumor finding at all
  - Any staging finding
  - Any pathologically staged finding
  - etc. 

Where I do see this impacting the ease of phenotyping, and by ease I once again mean not needing to list out every concept or use regex, is if the phenotype itself requires a specific subtype of finding. A notable use case would be limiting scope to _only include pathologically confirmed tumor findings_. These concepts are currently disconnected and spread out, but if we simply added a parent concept to each for C and P accordingly, then, for that example, it would simply be : _get descendants of the concept_id for the proposed "AJCC/UICC Pathological Finding"_ (more details on this specific piece in my most recent post above)

Another way to look at this would be that we are enabling that same functionality stated in the previous paragraph (parent classifier concepts to encapsulate the discrete list of acceptable concepts), and filling in the gaps as needed to distinguish:
- **Clinical vs. Pathological finding**
- **T vs. N vs. M findings**
- **Subcategories of tumor findings (staging concepts vs. grading concepts vs. dimension concepts, etc.)**
- AJCC/UICC version -- [this](https://athena.ohdsi.org/search-terms/terms/1634647) is already in place
- Generic grade (e.g. Grade 1 vs. 2 etc.) -- [this](https://athena.ohdsi.org/search-terms/terms/1634752) is already in place
- Generic stage (e.g. Stage 1 vs. Stage 2) -- [this](https://athena.ohdsi.org/search-terms/terms/1635838) is already in place
- Subgroups of the tumor finding classifications (e.g. FIGO **stage** finding) -- this is [somewhat already in place](https://athena.ohdsi.org/search-terms/terms/734316) but 1) does not distinguish between grade and stage and 2) **has no parent concept** to tie it in with the higher level staging and/or grading concepts)
  - Another less complex example (doesn't include both stage and grade) of this would be [Dukes' (staging) finding](https://athena.ohdsi.org/search-terms/terms/734328) which similarly **has no parent concept**

That was a larger response than I intended but would love your feedback on it, either here or during the discussion on this topic tomorrow.

**Comment by tseto** [comment]
The proposal to add high level concepts for T, N, M, Clinical and Path findings is a great idea. Instead of having one finding for T, which would bring in both clinical and pathologic T scores unless one uses the additional Clinical finding concept, what about creating high level concepts for clinical T, clinical N, clinical M, clinical stage, path T, path N, path M, path stage?  Or would the idea be to query on T, then filter again on clinical or pathologic?

**Comment by andysouth** [comment]
@rtmill @golozara thanks for the presentation yesterday & this proposal. Having the additional top level concepts would have helped me with an issue I had summarising our data a few months ago. 

I have been looking at visualising the omop hierarchies and wonder if these graphs for figo and ajcc are helpful. 

These ones take two steps from the relations table and I think help to see differences in the figo & ajcc hierachies. They could be used to represent the additional concept and relationships too. I've also done for 3 steps but they get a bit out of hand. 

[FIGO omop relations, 2 steps](https://github.com/OHDSI/OncologyWG/files/14886689/omopcept_graph-tree-figo-nts9-nta45-50x30cm.pdf)

[AJCC omop relations, 2 steps](https://github.com/OHDSI/OncologyWG/files/14886688/omopcept_graph-tree-ajcc-nts9-nta45-150x30cm.pdf)

These are created using the R package [omopcept](https://github.com/andysouth/omopcept) and I would welcome any suggestions. 

R Code to create (note that under development so may change)
```
remotes::install_github("andysouth/omopcept")
library(omopcept)

#these can take a few minutes
figo_rel2 <- omop_relations_recursive(734316, num_recurse=2)
ajcc_rel2 <- omop_relations_recursive(734320, num_recurse=2)

omop_graph(figo_rel2, ggrlayout="tree", legendshow = FALSE, saveplot = TRUE, width = 50, nodetxtangle = 45)
omop_graph(ajcc_rel2, ggrlayout="tree", legendshow = FALSE, saveplot = TRUE, width = 150, nodetxtangle = 45)
```

**Comment by rtmill** [comment]
@andysouth This is fantastic and couldn't have come at a better time. I'd just started exploring approaches to visualizing portions of the vocabularies for discussion purposes. 

Will certainly be checking/testing the package

Thanks for sharing

**Comment by kzollove** [comment]
@andysouth I've been searching for this exact package!!

**Comment by rtmill** [comment]
> @rtmill @golozara The proposal to add high level concepts for T, N, M, Clinical and Path findings is a great idea. But perhaps instead of having one finding for T, which would bring in both clinical and pathologic T scores unless one uses the additional Clinical finding concept, what about creating high level concepts for clinical T, clinical N, clinical M, clinical stage, path T, path N, path M, path stage? Or would the idea be to query on T, then filter again on clinical or pathologic?

It's a great point and I'll address this when I put together the mock vocab changes. We may run into conflicts in logic with that addition but can't be sure at this point
